src/NexusMods.App.UI/Pages/ILoadoutDataProvider.cs

```diff
@@ -22,6 +25,112 @@ public class LoadoutFilter
 
 public static class LoadoutDataProviderHelper
 {
+    public static ChangeSet<LoadoutItem.ReadOnly, EntityId> GetLinkedLoadoutItems(
+        IDb db,
+        LoadoutFilter loadoutFilter,
+        LibraryItemId libraryItemId)
+    {
+        List<EntityId> entityIds;
+
+        if (loadoutFilter.CollectionGroupId.HasValue)
+        {
+            entityIds = db.Datoms(
+                (LibraryLinkedLoadoutItem.LibraryItemId, libraryItemId),
+                (LoadoutItem.ParentId, loadoutFilter.CollectionGroupId.Value)
+            );
+        }
+        else
+        {
+            entityIds = db.Datoms(
+                (LibraryLinkedLoadoutItem.LibraryItemId, libraryItemId),
+                (LoadoutItem.LoadoutId, loadoutFilter.LoadoutId)
+            );
+        }
+
+        var changeSet = new ChangeSet<LoadoutItem.ReadOnly, EntityId>();
+
+        foreach (var entityId in entityIds)
+        {
+            var item = LoadoutItem.Load(db, entityId);
+            if (!item.IsValid()) continue;
+
+            changeSet.Add(new Change<LoadoutItem.ReadOnly, EntityId>(ChangeReason.Add, entityId, item));
+        }
```

Comment:
Any reason to prefer the for loop over Linq style here?
```suggestion
        var changeSet = new ChangeSet<LoadoutItem.ReadOnly, EntityId>(
            entityIds.Select(entityId => LoadoutItem.Load(db, entityId))
                .Where(item => item.IsValid())
                .Select(item => new Change<LoadoutItem.ReadOnly, EntityId>(ChangeReason.Add, item.Id, item))
        );
```